course_management.lua

```diff
@@ -170,59 +174,69 @@ function courseplay:loadCourse(vehicle, id, useRealId, addCourseAtEnd) -- fn is
 				end;
 				courseplay:debug(string.format('course 1 has %d crossing points (excluding first point), course 2 has %d crossing points (excluding last point), useFirstMatch=%s', #crossingPoints[1], #crossingPoints[2], tostring(useFirstMatch)), 8);
 
-				--find < 50m match
-				local smallestDist = math.huge;
+				--find < wpDistMax match with lowest total turn angle
+				local smallestAngle, smallestDist = math.huge, math.huge;
 				if #crossingPoints[1] > 0 and #crossingPoints[2] > 0 then
 					for _,wpNum1 in pairs(crossingPoints[1]) do
 						local wp1 = course1[wpNum1];
 						for _,wpNum2 in pairs(crossingPoints[2]) do
 							local wp2 = course2[wpNum2];
 							local dist = courseplay:distance(wp1.cx, wp1.cz, wp2.cx, wp2.cz);
-							courseplay:debug(string.format('course1 wp %d, course2 wp %d, dist=%s', wpNum1, wpNum2, tostring(dist)), 8);
-							if dist and dist ~= 0 and dist < 50 then
-								if useFirstMatch then
+							--Calculate actual turn direction between pair of crosspoints
+							local angleTurn = math.atan2(wp2.cx-wp1.cx,wp2.cz-wp1.cz) -- in radians 
+							--add direction change differences between original direction and turn direction and destination direction							
+							local totalAngle=math.deg(
+								   math.abs(getDeltaAngle(math.rad(wp1.angle),angleTurn)) + 
+								   math.abs(getDeltaAngle(angleTurn,math.rad(wp2.angle))))
+							angleTurn = math.deg(angleTurn) -- now in degrees
+							--courseplay:debug(string.format('course1 wp %d, course2 wp %d, dist=%s', wpNum1, wpNum2, tostring(dist)), 8);
+							if dist and dist ~= 0 and dist < wpDistMax then
+								courseplay:debug(string.format('wp1 %d %.2f° wp2 %d %.2f° dist=%.1f angleTurn %.2f°, totalAngle %.2f°, lowA %.2f°, lowD %.1f', 
```

Comment:
Update this to `courseplay.debugVehicle(8, vehicle,'wp1 %d %.2f° wp2 %d %.2f° dist=%.1f angleTurn %.2f°, totalAngle %.2f°, lowA %.2f°, lowD %.1f', wpNum1, wp1.angle, wpNum2, wp2.angle, dist, angleTurn , totalAngle, smallestAngle, smallestDist)` thanks were trying to update the debug to that function